### PR TITLE
Unified metastore interface

### PR DESCRIPTION
**Pull request by jaime0815**
issue #17041 

  Co-authored-by:  kejiang <ke.jiang@zilliz.com>
  Co-authored-by:  yun.zhang <yun.zhang@zilliz.com>

Signed-off-by:  yun.zhang <yun.zhang@zilliz.com>

### CODE REVIEW

File: internal/metastore/kv/suffix_snapshot.go [hunk 1]
```diff
@@ -14,7 +14,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package rootcoord
+package kv
```
Comment (xiaofan-luan) [inline internal/metastore/kv/suffix_snapshot.go:17]: We will need to think of suffix snapshot clean up later

File: internal/rootcoord/meta_table.go [hunk 2]
```diff
@@ -126,225 +105,113 @@ func NewMetaTable(txn kv.TxnKV, snap kv.SnapShotKV) (*MetaTable, error) {
...
-		}
-		meta[k] = string(v)
+	coll.CreateTime = ts
+	for _, partition := range coll.Partitions {
+		partition.PartitionCreatedTimestamp = ts
 	}
 
-	// save ddOpStr into etcd
+	meta := map[string]string{}
 	meta[DDMsgSendPrefix] = "false"
```
Comment (xiaofan-luan) [inline internal/rootcoord/meta_table.go:184]: remove these weird logic, make the DDL idempotent and reriable

File: internal/metastore/kv/suffix_snapshot.go (hunk 1 above)
Comment (jaime0815) [reply internal/metastore/kv/suffix_snapshot.go:17]: We will support cleaning up snapshot data from mysql metastore with high priority

**Comment by xiaofan-luan** [comment]
@jaime0815 ready to be reviewed on that?

**Comment by jaime0815** [comment]
> @jaime0815 ready to be reviewed on that?

Yes

**Pull request merged**